src/main/java/com/jwplayer/southpaw/util/ByteArraySet.java

```diff
@@ -312,10 +312,16 @@ public ByteArray next() {
             int size = (int) currentChunk.bytes[currentOffset];
             if(size == 0) {
                 currentOffset += 1;
+                updateState();
                 return null;
             }
             byte[] retVal = Arrays.copyOfRange(currentChunk.bytes, currentOffset + 1, currentOffset + size + 1);
             currentOffset += 1 + size;
+            updateState();
```

Comment:
This state update is mandatory, it leads to data loss otherwise

Reply:
sounds like an important gotcha.  Leave this comment above within a code comment?